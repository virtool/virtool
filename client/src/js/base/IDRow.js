import React from "react";
import { connect } from "react-redux";

/**
 * An component for easily rendering unique ID table rows when the ``show_ids`` account setting is on. This component
 * would be used for example in a table on a sample detail page to show the database id of the sample.
 *
 * @param id
 * @param showIds
 */
export const IDRowComponent = ({ id, showIds }) => (
    showIds ? (
        <tr>
            <th>Unique ID</th>
            <td>{id}</td>
        </tr>
    ) : null
);

const mapStateToProps = (state) => ({
    showIds: state.account.settings.show_ids
});

export const IDRow = connect(mapStateToProps)(IDRowComponent);
