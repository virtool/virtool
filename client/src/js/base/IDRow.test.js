import { IDRow, IDRowComponent } from "./IDRow";

describe("<IDRow />", () => {
    let props;
    let wrapper;

    it("renders connected component", () => {
        const initialState = {
            account: {
                settings: {
                    show_ids: true
                }
            }
        };
        const store = mockStore(initialState);

        const container = shallow(<IDRow store={store} />);

        expect(container).toMatchSnapshot();
    });

    it("when [account.settings.show_ids=false], renders nothing (null)", () => {
        props = { id: "test_id", showIds: false };
        wrapper = shallow(<IDRowComponent {...props} />);

        expect(wrapper.html()).toBe(null);
        expect(wrapper).toMatchSnapshot();
    });

    describe("when [account.settings.show_ids=true]: ", () => {
        
        beforeEach(() => {
            props = { id: "test_id", showIds: true };
            wrapper = mount(
                <table>
                    <tbody>
                        <IDRowComponent {...props} />
                    </tbody>
                </table>
            );
        });

        it("renders correctly", () => {
            expect(wrapper).toMatchSnapshot();
        });

        it("renders a table row with a header and the target's unique database id", () => {
            
            expect(wrapper.find('tr').exists()).toBe(true);
            expect(wrapper.find('th').text()).toEqual("Unique ID");
            expect(wrapper.find('td').text()).toEqual(props.id);
        });

    });

});
