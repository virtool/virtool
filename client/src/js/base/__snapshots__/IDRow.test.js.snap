// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IDRow /> renders connected component 1`] = `
<IDRowComponent
  dispatch={[Function]}
  showIds={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`<IDRow /> when [account.settings.show_ids=false], renders nothing (null) 1`] = `""`;

exports[`<IDRow /> when [account.settings.show_ids=true]:  renders correctly 1`] = `
<table>
  <tbody>
    <IDRowComponent
      id="test_id"
      showIds={true}
    >
      <tr>
        <th>
          Unique ID
        </th>
        <td>
          test_id
        </td>
      </tr>
    </IDRowComponent>
  </tbody>
</table>
`;
